From 97741cd6bd1591e1310ca6ba9ce821b389abe5a7 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <baimingcong@uniontech.com>
Date: Fri, 28 Jun 2024 15:08:59 +0800
Subject: [PATCH 090/360] irqchip: adapt irq-gic-phytium-2500-its.c to >= 6.7

A previous change by Kirill A. Shutemov <kirill.shutemov@linux.intel.com>
renamed MAX_ORDER to MAX_PAGE_ORDER:

    mm, treewide: rename MAX_ORDER to MAX_PAGE_ORDER

    commit 23baf831a32c ("mm, treewide: redefine MAX_ORDER sanely") has
    changed the definition of MAX_ORDER to be inclusive.  This has caused
    issues with code that was not yet upstream and depended on the previous
    definition.

    To draw attention to the altered meaning of the define, rename MAX_ORDER
    to MAX_PAGE_ORDER.

    Link: https://lkml.kernel.org/r/20231228144704.14033-2-kirill.shutemov@linux.intel.com
    Signed-off-by: Kirill A. Shutemov <kirill.shutemov@linux.intel.com>
    Cc: Linus Torvalds <torvalds@linux-foundation.org>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>

Signed-off-by: Mingcong Bai <baimingcong@uniontech.com>
---
 drivers/irqchip/irq-gic-phytium-2500-its.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/irqchip/irq-gic-phytium-2500-its.c b/drivers/irqchip/irq-gic-phytium-2500-its.c
index ab55370ab01d..8e72c401ff98 100644
--- a/drivers/irqchip/irq-gic-phytium-2500-its.c
+++ b/drivers/irqchip/irq-gic-phytium-2500-its.c
@@ -2492,8 +2492,8 @@ static bool its_parse_indirect_baser(struct its_node *its,
 	 * feature is not supported by hardware.
 	 */
 	new_order = max_t(u32, get_order(esz << ids), new_order);
-	if (new_order >= MAX_ORDER) {
-		new_order = MAX_ORDER - 1;
+	if (new_order >= MAX_PAGE_ORDER) {
+		new_order = MAX_PAGE_ORDER - 1;
 		ids = ilog2(PAGE_ORDER_TO_SIZE(new_order) / (int)esz);
 		pr_warn("ITS@%pa: %s Table too large, reduce ids %llu->%u\n",
 			&its->phys_base, its_base_type_string[type],
-- 
2.20.1

